src/Storages/Streaming/benchmarks/atomic_lock_double_check_perf.cpp

```diff
@@ -0,0 +1,145 @@
+#include <benchmark/benchmark.h>
+
+#include <Poco/Mutex.h>
+
+#include <atomic>
+#include <list>
+#include <memory>
+#include <mutex>
+#include <thread>
+
+
+class ComplexSharedResource
+{
+    [[maybe_unused]] std::list<int> lst;
+    [[maybe_unused]] std::list<int>::iterator itor;
+    [[maybe_unused]] int valid_num;
+};
+
+static void BM_ScopedLockOnly(benchmark::State & state)
```

Comment:
Consider anonymous namespace instead of static

Reply:
like 

```
namespace
{
..
}

BENCHMARK(BM_ScopedLockOnly)->MinTime(3);
BENCHMARK(BM_AtomicWithScopedLockOnPocoFastMutex)->MinTime(3);
```

Reply:
i see. tks